NET01.2/Matriсes/SquareMatrix.cs

```diff
@@ -0,0 +1,172 @@
+            }
+            set
+            {
+                if (size >= 0)
+                {
+                    array = new T[value * value];
+                    size = value;
+                }
+                else
+                    throw new Exception("Size of matrix can't be negative.");
+            }
+        }
+        /// <summary>
+        /// Matrix array access indexer
+        /// </summary>
+        /// <param name="indexI"></param>
+        /// <param name="indexJ"></param>
+        /// <returns>Element by index</returns>
+        /// <sets>Value by index</sets>
+        public T this[int indexI, int indexJ]
+        {
+            get
+            {
+                if (indexI >= 0 && indexI < size && indexJ >= 0 && indexJ < size)
+                {
+                    int indexOfElement = 0;
+                    for (int i = 0; i < size; i++)
+                    {
+                        for (int j = 0; j < size; j++)
+                        {
+                            if (i == indexI && j == indexJ)
+                            {
+                                break;
+                            }
+                            indexOfElement++;
+                        }
+                    }
+                    return array[indexOfElement];
+                }
+                else
+                    throw new Exception("Invalid element index");
+            }
+            set
+            {
+                if (indexI >= 0 && indexI < size && indexJ >= 0 && indexJ < size)
+                {
+                    int indexOfElement = 0;
+                    for (int i = 0; i < size; i++)
+                    {
+                        for (int j = 0; j < size; j++)
+                        {
+                            if (i == indexI && j == indexJ)
+                            {
+                                T item1 = array[indexOfElement];
+                                T item2 = value;
+                                if(!EqualityComparer<T>.Default.Equals(item1, item2))
+                                {
+                                    Notify += ChangeValue;
+                                    Notify(i, j, array[indexOfElement]);
+                                    Notify -= ChangeValue;
+                                    array[indexOfElement] = value;
+                                    break;
+                                }
+                            }
+                            indexOfElement++;
+                        }
+                    }
+                }
+                else
+                    throw new Exception("Invalid element index");
+            }
+        }
+        /// <summary>
+        /// Ovveriden method ToString()
+        /// </summary>
+        /// <returns>String representation of a matrix</returns>
+        public override string ToString()
+        {
+            int sizeOfMatrix = Size;
+            string outputMatrix = String.Empty;
```

Comment:
for(int i=0; i < size; i++)
{
for(int j=0; j < size; j++)
{
stringBuilder.Append(array[i*size + j]);
}
str
}